framework/Optimizers/RavenSampled.py

```diff
@@ -0,0 +1,566 @@
+    """
+
+  @abc.abstractmethod
+  def _updateConvergence(self, traj, rlz, old, acceptable):
+    """
+      Updates convergence information for trajectory
+      @ In, traj, int, identifier
+      @ In, acceptable, str, condition of point
+      @ Out, converged, bool, True if converged on ANY criteria
+    """
+
+  @abc.abstractmethod
+  def _updatePersistence(self, traj, converged, optVal):
+    """
+      Update persistence tracking state variables
+      @ In, traj, identifier
+      @ In, converged, bool, convergence check result
+      @ In, optVal, float, new optimal value
+      @ Out, None
+    """
+
+  @abc.abstractmethod
+  def _rejectOptPoint(self, traj, info, old):
+    """
+      Having rejected the suggested opt point, take actions so we can move forward
+      @ In, traj, int, identifier
+      @ In, info, dict, meta information about the opt point
+      @ In, old, dict, previous optimal point (to resubmit)
+    """
+
+  def _updateSolutionExport(self, traj, rlz, acceptable):
+    """
+      Stores information to the solution export.
+      @ In, traj, int, trajectory which should be written
+      @ In, rlz, dict, collected point
+      @ In, acceptable, bool, acceptability of opt point
+      @ Out, None
+    """
+    # make a holder for the realization that will go to the solutionExport
+    toExport = {}
+    # add some meta information
+    toExport.update({'iteration': self.getIteration(traj),
+                     'trajID': traj,
+                     'accepted': acceptable,
+                    })
+    # optimal point input and output spaces
+    objValue = rlz[self._objectiveVar]
+    if self._minMax == 'max':
+      objValue *= -1
+    toExport[self._objectiveVar] = objValue
+    toExport.update(self.denormalizeData(dict((var, rlz[var]) for var in self.toBeSampled)))
+    # constants and functions
+    toExport.update(self.constants)
+    toExport.update(dict((var, rlz[var]) for var in self.dependentSample))
+    # additional from from inheritors
+    toExport.update(self._addToSolutionExport(traj, rlz, acceptable))
+    # formatting
+    toExport = dict((var, np.atleast_1d(val)) for var, val in toExport.items())
+    self._solutionExport.addRealization(toExport)
+
+  def _addToSolutionExport(self, traj, rlz, acceptable):
+    """
+      Contributes additional entries to the solution export.
+      Should be used by inheritors instead of overloading updateSolutionExport
+      @ In, traj, int, trajectory which should be written
+      @ In, rlz, dict, collected point
+      @ In, acceptable, bool, acceptability of opt point
+      @ Out, toAdd, dict, additional entries
+    """
+    return {}
+
+  # END resolving potential opt points
+  # * * * * * * * * * * * * * * * *
+
+  def _cancelAssociatedJobs(self, traj, step=None):
+    """
+      Queues jobs to be cancelled based on opt run
+      @ In, traj, int, trajectory identifier
+      @ In, step, int, optional, iteration identifier
```

Comment:
should we explain what happens if it is not inputted (if it is None)?

Reply:
I guess, it's just not used if not provided, but I can say that.